dali/kernels/imgproc/convolution/cutlass/conv_window_configuration.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_CONVOLUTION_CUTLASS_CONV_WINDOW_CONFIGURATION_H_
+#define DALI_KERNELS_IMGPROC_CONVOLUTION_CUTLASS_CONV_WINDOW_CONFIGURATION_H_
+
+#include "cutlass/cutlass.h"
+
+#include "dali/core/span.h"
+
+namespace cutlass {
+namespace gemm {
+
+template <int TotalAlignedSize, bool InnerConv, bool UseSharedMem = true>
+struct ConvWindowConfiguration {
+  static int const kTotalAlignedSize = TotalAlignedSize;
+  static bool const kInnerConv = InnerConv;
+  static bool const kUseSharedMem = UseSharedMem;
+  static_assert(kUseSharedMem, "Reading window directly from GMEM is not yet implemented");
+  static_assert((!kInnerConv && kTotalAlignedSize % 4 == 0) ||
+                    (kInnerConv && kTotalAlignedSize % 2 == 0),
+                "The total window size needs to be divisible for alignment purposes");
+  static int const kWindowDecreasingCenter =
+      kInnerConv ? kTotalAlignedSize / 2 : kTotalAlignedSize / 4;
+  static int const kWindowIncreasingCenter = kInnerConv ? -1 : (kTotalAlignedSize / 4) * 3;
+
+  template <typename T>
+  using PaddedWindowBuffer = dali::span<T, ConvWindowConfiguration::kTotalAlignedSize>;
+
+  template <bool mirrored>
+  CUTLASS_HOST_DEVICE constexpr static int getWindowCenter() {
+    if (mirrored) {
+      return kWindowDecreasingCenter;
+    }
+    return kWindowIncreasingCenter;
+  }
+
+  static int const kMaxWindowRadiusSpan =
+      kInnerConv ? kTotalAlignedSize / 2 : kTotalAlignedSize / 4;
+
+  template <typename T>
+  static void prepare_window(PaddedWindowBuffer<T> dst, dali::span<const T> src,
+                             int num_channels = 1) {
+    int radius = src.size() / 2;
+    memset(dst.data(), 0, sizeof(T) * kTotalAlignedSize);
```

Comment:
Why do we need to memset it?

Reply:
To make sure that the memory is zeroed-out. When we process with channels, we need the zeroes in there for the non-current channels.

Reply:
OK